components/esp_modem/include/cxx_include/esp_modem_dte.hpp

```diff
@@ -116,8 +116,8 @@ class DTE : public CommandableIf {
     Lock internal_lock{};                                   /*!< Locks DTE operations */
     unique_buffer buffer;                                   /*!< DTE buffer */
     std::shared_ptr<CMux> cmux_term;                        /*!< Primary terminal for this DTE */
-    std::shared_ptr<Terminal> command_term;                 /*!< Reference to the terminal used for sending commands */
-    std::shared_ptr<Terminal> data_term;                    /*!< Secondary terminal for this DTE */
+    std::shared_ptr<Terminal> term;                         /*!< Reference to the primary terminal (mostly for sending commands) */
```

Comment:
> Does it make sense to convert this to a std:shared_ptr<std::vector<Terminal>> ?

I think it does if we make the CMUX mode really flexible (i.e. allowing for creation of an arbitrary number of terminals), but we would have to come up with a compact API how to access different terminals. Currently we run commands on primary and data operations on secondary terminal.

I'd suggest that we rename the terminals per your recommendation and keep the multi-terminal API for future.

Reply:
Yes, the usage of `std::vector` would allow a more flexible CMUX mode and I agree that could be for the future. 

About the renaming, is my understanding that it's possible to switch their role and send data on primary and commands on secondary. Am I correct?